creator-node/src/services/stateMachineManager/stateReconciliation/updateReplicaSet.jobProcessor.ts

```diff
@@ -661,17 +668,36 @@ const _canReconfig = async ({
       )
     }
 
-    canReconfig =
+    // Reconfig is necessary if endpoint doesn't exist in mapping because this means the node was deregistered
+    const isAnyNodeInReplicaSetDeregistered =
+      !oldPrimarySpId || !oldSecondary1SpId || !oldSecondary2SpId
+    if (isAnyNodeInReplicaSetDeregistered) {
+      return {
+        canReconfig: true,
+        currentPrimarySpId,
```

Comment:
can we rename these variables to chain-*?

Reply:
for sure, that's a bit clearer